utils/handleToasts.ts

```diff
@@ -1,49 +1,57 @@
 import { toast } from 'react-toastify';
 
-function generateToastID() {
-  return checkIsIdOccupied(randomId());
-}
-
-function randomId() {
-  return Math.floor(Math.random() * 100000000).toString();
-}
-
-function checkIsIdOccupied(toastId: string) {
-  if (document.getElementById(toastId.toString()) !== null) {
-    return checkIsIdOccupied(randomId());
-  }
-  return toastId;
-}
-
 export function handleErrorToast(errorText: string) {
   try {
-    const generatedToastID = generateToastID();
-    if (!document) {
-      return;
-    }
-
-    toast(errorText, {
+    const toastId = toast(errorText, {
       type: 'error',
-      toastId: generatedToastID,
-      onOpen: () => {
-        setTimeout(() => {
-          document
-            .getElementById(generatedToastID)
-            ?.addEventListener('mouseenter', () => {
-              toast.update(generatedToastID, {
-                progress: 1,
+      toastId: window.crypto.randomUUID(),
+    });
+
+    const observer = new MutationObserver((mutationsList) => {
+      for (const mutation of mutationsList) {
+        mutation.addedNodes.forEach((node) => {
+          // check if node is an HTMLElement which unlock a usage of node.children
+          if (node instanceof HTMLElement) {
+            // if there are more than 1 toast, this should be working
+            if (node.id === toastId.toString()) {
```

Comment:
Rivi 14:n tarkoitus on siinä, jotta saadaan tieto `node.children`:in ja itseasiassa `.id`:n olemassa olosta myös. Lisään sen kommenttiin myös 

Rivi 16. Jos Toasti **EI** ole ensimmäinen, `node.id === toastId.toString()` toimii. Jos toasti on ensimmäinen, joudutaan käyttämään alla olevaa For-looppia. Syy tähän näyttäisi olevan se, että ensimmäisen Toastin kohdalla `node` on `Toast container` ei itse Toast.

En oikein osaa selittää, mutta sen pystyy helposti testaamaan, kun lisää vain vaikka riville 13 `console.log(node)` :smile:

Reply:
Jep ekalla kertaa luodaan tuoas container jota ei poisteta, siksi 1. ja muut kerrat poikkeavat.